File: tests/fixtures/small_python_app/app/services/user_service.py
"""Business logic for users."""
from app.repositories.user_repository import UserRepository

_repo = UserRepository()


def create_user(payload):
    if not payload.get("name"):
        raise ValueError("name is required")
    return _repo.insert(payload)


def get_user(user_id):
    return _repo.find_by_id(user_id)

File: tests/fixtures/node_api_app/src/services/userService.js
const config = require('../config');

const users = new Map();
let nextId = 1;

function createUser(payload) {
  if (!payload || !payload.name) {
    throw new Error('name is required');
  }
  const user = { id: nextId, name: payload.name };
  nextId += 1;
  users.set(user.id, user);
  return user;
}

function getUser(id) {
  return users.get(id) || null;
}

module.exports = { createUser, getUser };
